src/databricks/sql/auth/retry.py

```diff
@@ -362,6 +363,12 @@ def should_retry(self, method: str, status_code: int) -> Tuple[bool, str]:
         if status_code == 501:
             raise NonRecoverableNetworkError("Received code 501 from server.")
 
+        if self.command_type == CommandType.FETCH_RESULTS_ORIENTATION_FETCH_NEXT:
+            return (
+                False,
+                "FetchResults with FETCH_NEXT orientation are not idempotent and is not retried",
+            )
+
```

Comment:
It is not idempotent only in INLINE mode

Reply:
@jackyhu-db Updated the error message

Reply:
my point is it should return false (stop retrying) if if it NOT INLINE mode.

Reply:
@jackyhu-db The issue is that in INLINE FETCH_NEXT is causing the issue and we don't want to retry. Why stop retrying when in NOT INLINE ? Anyways not INLINE uses cloudfetch which does not use FetchResults call